client/server/middleware/geoip-header.ts

```diff
@@ -0,0 +1,49 @@
+import { type RequestHandler } from 'express';
+
+/**
+ * Adds the `x-geoip-country-code` header to the request. This is used to simulate the header added
+ * in production environments, which is not available in all environments.
+ * @returns A middleware function that adds the `x-geoip-country-code` header to the request.
+ */
+export default function (): RequestHandler {
+	// To avoid delays on handling individual requests, fetch the geolocation once at server
+	// startup. This will retrieve the geolocation of the server (not the client), but it's
+	// expected that the middleware would be enabled on a developer's own machine.
+	const getGeoCountryCode = ( () => {
+		let countryCode: string | undefined;
+		let countryCodePromise: Promise< string | undefined > | undefined;
+
+		return async () => {
+			if ( ! countryCode ) {
+				if ( ! countryCodePromise ) {
+					countryCodePromise = fetch( 'https://public-api.wordpress.com/geo/' )
+						.then( async ( response ) => response.json() )
+						.then( ( { country_short } ) => country_short )
+						// Support offline development and set an expectation that the header is not
+						// always available by quietly skipping if the request fails.
+						.catch( () => {
+							// Reset the promise so we can try again on the next request
+							countryCodePromise = undefined;
+							return undefined;
+						} );
+				}
+
+				countryCode = await countryCodePromise;
+			}
+
+			return countryCode;
+		};
+	} )();
```

Comment:
Can we remove the IIFE around this code and do the work directly in the middleware function? The inner async function can be the new `getGeoCountryCode`. The IIFE doesn't seem to do anything valuable, except hiding the `countryCode` variable.

Reply:
> Can we remove the IIFE around this code and do the work directly in the middleware function? The inner async function can be the new `getGeoCountryCode`. The IIFE doesn't seem to do anything valuable, except hiding the `countryCode` variable.

Sure, updated in 3f7b0e4d9d2. I was trying to use IIFE to keep this self-contained, but I agree it creates some unnecessary confusion.